novice-goals.Rmd

```diff
@@ -0,0 +1,226 @@
+# Outline {#outline}
+
+This outline describes the questions that the novice courses on R and Python will answer.
+The advanced course can then assume that learners have hands-on experience with these topics
+but nothing more.
+
+## Getting started
+
+-   What are the different ways I can interact with software?
+    -   console
+    -   scripts
+-   How can I find and view help?
+    -   In the IDE
+    -   Stack Overflow
+-   How can I inspect data while I'm working on it?
+    -   table viewers
+-   Where does stuff actually live on my computer?
+    -   directory structure on Windows and Unix
+    -   absolute vs. relative paths
+    -   data on disk vs. data in memory
+
+## Data manipulation
+
+-   How can I read tabular data into a program?
+    -   what CSV is, where it comes from, and why people use it
+    -   reading files
+-   How can I select subsets of my data?
+    -   select
+    -   filter
+    -   arrange
+    -   Boolean conditions
+-   How can I calculate new values?
+    -   mutate
+    -   ifelse
+-   How can I tell what's gone wrong in my programs?
+    -   reading error messages
+    -   the difference between syntax and runtime errors
+-   How can I operate on subsets of my data?
+    -   group
+    -   summarize
+    -   split-apply-combine
+-   How can I work with two or more datasets?
+    -   join
+-   How can I save my results?
+    -   writing files
+-   What *isn't* included?
+    -   anything other than reasonably tidy tabular data
+    -   map
```

Comment:
I actually think that, while mapping as a topic is difficult, the implementation (at least in R) is relatively straight forward using purrr. Plus by at least showing it, the learners can see that this tool exists. And then they won't be tempted to using for loops (not good in R).

Reply:
I will defer to the R crew - @cwickham @joelostblom ? Do you think learners will get far enough to need `map` given everything else on the table?

Reply:
While I'm flattered to be included in the R crew, @lwjohnst86 is much more knowledgeable here and can answer this better than I can.